chore: adding FAQs for multi-engine (#3761)
open-policy-agent/gatekeeper: website/docs/validating-admission-policy.md

```

To see this in action, checkout this [demo](https://github.com/open-policy-agent/gatekeeper/tree/master/demo/scoped-enforcement-actions)

## FAQs

<details>

> <summary>Do all engines in a ConstraintTemplate are evaluated? Is there a fallback among engines?</summary>

Reviewer: ```suggestion
<summary>Do all engines in a ConstraintTemplate get evaluated? Is there a fallback among engines?</summary>
```

> Only one engine is evaluated for each ConstraintTemplate. `K8sNativeValidation` engine holds a higher priority than the `Rego` engine. There is no fallback mechanism between engines, hence a logical/syntactical error in the policy logic is treated as violation depending on the enforcement action specified in the Constraint.

Reviewer: ```suggestion
Only one engine is evaluated for each ConstraintTemplate. The `K8sNativeValidation` engine holds a higher priority than the `Rego` engine. There is no fallback mechanism between engines, hence a logical/syntactical error in the policy logic is treated as a violation depending on the enforcement action specified in the Constraint.
```

> Only one engine is evaluated for each ConstraintTemplate. K8sNativeValidation engine hold higher priority than Rego engine. There is no fallback mechanism between engines, hence a logical/syntactical error in policy logic is treated as violation depending on the enforcement action set of the Constraints created for ConstraintTemplate.

Reviewer: ```suggestion
Only one engine is evaluated for each ConstraintTemplate. `K8sNativeValidation` engine holds a higher priority than the `Rego` engine. There is no fallback mechanism between engines, hence a logical/syntactical error in the policy logic is treated as violation depending on the enforcement action specified in the Constraint.
```

</details>

<details>

> <summary>If I have a template with Rego and CEL, which policy logic will be used in evaluating resources?</summary>

Reviewer: ```suggestion
<summary>If I have a template with Rego and CEL, which policy engine will be used when evaluating resources?</summary>
```

> K8sNativeValidation engine holds a higher priority than the Rego engine, so with a ConstraintTemplate that has both Rego and CEL. Policy logic written in CEL will get evaluated by the K8sNativeValidation engine.

Reviewer: ```suggestion
The K8sNativeValidation engine holds a higher priority than the Rego engine. If a ConstraintTemplate has both Rego and CEL, then the policy logic written in CEL will get evaluated by the K8sNativeValidation engine.
```

</details>

<details>
<summary>Can I change the priority of engines per ConstraintTemplate?
</summary>

> Engine priority cannot be set per ConstraintTemplate.

Reviewer: ```suggestion
No, engine priority cannot be modified.
```

Removing "per ConstraintTemplate" since it makes it look like it can be done but just not per CT.

Author: Makes sense
